### PR TITLE
[Release] sample strategy, testing, bugfix #minor

### PR DESCRIPTION
**Pull request by dema-trading-ai**
# In this release:

- Bugfixes, 
- README additions, 
- Dynamic ROI
- Unit testing setup + CI 
- Sample Strategy
- Fee implementation

### CODE REVIEW

File: data/tradingmodule.py [hunk 1]
```diff
@@ -148,11 +151,28 @@ def check_roi_open_trade(self, trade: Trade, ohlcv: Series) -> bool:
...
         :rtype: boolean
         """
-        if trade.profit_percentage > float(self.config['roi']):
+        time_passed = datetime.fromtimestamp(ohlcv['time'] / 1000) - trade.opened_at
+        if trade.profit_percentage > self.calculate_roi_over_time(time_passed):
             self.close_trade(trade, reason="ROI", ohlcv=ohlcv)
             return True
         return False
 
+    def calculate_roi_over_time(self, time_passed) -> float:
```
Comment (pvdklei) [inline data/tradingmodule.py:160]: I feel like the function name is a little misleading, it does not calculate roi but finds the appropriate config value

**Review by pvdklei — COMMENTED** [review]
Got some very minor comments, overall it looks ready for shipping!

File: backtesting/strategy.py [hunk 2]
```diff
@@ -1,63 +1,63 @@
...
         :type indicators: DataFrame
-        :param current_candle: Last candle
+        :param current_candle: Last candle filled with indicators from generate_indicators
         :type current_candle: Series
-        :return: Dataframe filled with buy signals
-        :rtype: DataFrame
+        :return: Current candle filled with buy signals
+        :rtype: Series
         """
-        pass
```
[the remark this answers is not in the document]
Comment (noudcorten) [reply backtesting/strategy.py:42]: Good remark. Not sure.

[the remark this answers is not in the document]
Comment (dema-trading-ai) [reply backtesting/strategy.py:42]: @noudcorten you want this changed in this version?

File: data/tradingmodule.py (hunk 1 above)
Comment (dema-trading-ai) [reply data/tradingmodule.py:160]: @pvdklei good remark. Wil refactor

File: backtesting/strategy.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (noudcorten) [reply backtesting/strategy.py:42]: @dema-trading-ai @pvdklei I think the `pass` argument is no strong requirement for the abstract class. I found [this](https://stackoverflow.com/questions/4382945/abstract-methods-in-python) in which in they use `return` as well. So no need to change this.

**Review by marijn111 — APPROVED** [review]

**Review by marijn111 — DISMISSED** [review]

File: config.json [hunk 3]
```diff
@@ -7,10 +7,18 @@
...
+    "120": 3
+  },
+  "pairs": [
+    "BTC/USDT",
+    "ETH/USDT",
+    "LTC/USDT"
+  ],
   "currency": "USDT",
-  "fee": null,
+  "fee": -1.4,
```
Comment (noudcorten) [inline config.json:21]: Why is the fee negative here? (@dema-trading-ai)

**Pull request merged**